### DISCUSSION TITLE
Modulation nodes

### DISCUSSION THREAD

**Discussion by rihardsgravis** [chaosprint/glicol#145] [Ideas]
Hello there 🖖 First of all, thanks for creating Glicol, it's awesome 👍 

First here's a simplified, minimal example to understand more what I mean. So instead of:
```
~sound_source: sin 440 >> mul ~envelope;

~envelope: imp 2.0 >> envperc 0.001 0.1;

out: ~sound..;
```

I would love to write:
```
~sound_source: sin 440 >> ~envelope;

// This line needs some changes that are currently not there
~envelope: imp 2.0 >> envperc 0.001 0.1;

out: ~sound..;
```

Meaning – it would be nice if I could reference chain nodes only by their name, without knowing what that node actually does. This would allow quick switching between various types of nodes with the same name by commenting them in/out + without changing the target chain where they are actually used.

Could maybe chains "to be used for modulating another signal chain" to be defined with some other prefix or other custom formatting and once referenced would magically understand that they need to `mul` the incoming signal.

Does something like this make sense? Maybe there is already an approach one could use to achieve this?

**Comment by chaosprint** [comment]
In this issue(https://github.com/chaosprint/glicol/issues/107) you can see `~ping` is used as a "ref node"
Under the hood, the engine will connect the end of that chain of signal "~ping" to the next node `add ~delay`.
So this ref node syntax is retained for this kind of usage.

On the other hand, if you want:

```
~sound_source: sin 440 >> ~envelope;

// This line needs some changes that are currently not there
~envelope: imp 2.0 >> envperc 0.001 0.1;

out: ~sound..;
```

It will mean to connect `sin 440` to the start of `~envelope`, which may cause some confusion.
